GitUI/CommandsDialogs/RevisionDiff.cs

```diff
@@ -603,5 +615,78 @@ private void saveAsToolStripMenuItem1_Click(object sender, EventArgs e)
                 }
             }
         }
+
+
+        private bool DeleteSelectedDiffFiles()
+        {
+            if (DiffFiles.Focused)
+            {
+                return DeleteSelectedFiles();
+            }
+            return false;
+        }
+
+        private bool DeleteSelectedFiles()
+        {
+            try
+            {
+                if (DiffFiles.SelectedItem == null || !DiffFiles.Revision.IsArtificial() ||
+                    MessageBox.Show(this, _deleteSelectedFiles.Text, _deleteSelectedFilesCaption.Text, MessageBoxButtons.YesNo) !=
+                    DialogResult.Yes)
+                {
+                    return false;
+                }
+
+                var selectedItems = DiffFiles.SelectedItems;
+                if (DiffFiles.Revision.Guid == GitRevision.IndexGuid)
+                {
+                    /// <summary>
+                    var files = new List<GitItemStatus>();
+                    var stagedItems = selectedItems.Where(item => item.IsStaged);
+                    foreach (var item in stagedItems)
+                    {
+                        if (!item.IsNew)
+                        {
+                            Module.UnstageFileToRemove(item.Name);
+
+                            if (item.IsRenamed)
+                                Module.UnstageFileToRemove(item.OldName);
+                        }
+                        else
+                        {
+                            files.Add(item);
+                        }
+                    }
+                    Module.UnstageFiles(files);
+                }
+                DiffFiles.StoreNextIndexToSelect();
+                var items = DiffFiles.SelectedItems.Where(item => !item.IsSubmodule);
+                foreach (var item in items)
+                {
+                    File.Delete(Path.Combine(Module.WorkingDir, item.Name));
+                }
+                //TBD RefreshRevisions();
+            }
+            catch (Exception ex)
+            {
+                MessageBox.Show(this, _deleteFailed.Text + Environment.NewLine + ex.Message);
+                return false;
+            }
+            return true;
+        }
+
+        private void diffDeleteFileToolStripMenuItemClick(object sender, EventArgs e)
+        {
+            DeleteSelectedFiles();
+        }
+
+        private void diffEditFileToolStripMenuItem_Click(object sender, EventArgs e)
+        {
+            var item = DiffFiles.SelectedItem;
+            var fileName = Path.Combine(Module.WorkingDir, item.Name);
```

Comment:
there are quite a few calls to `Path.Combine(Module.WorkingDir, item.Name)` - that's a good candidate for refactor
perhaps something like `RevisionDiffController.GetFullFilePath(string selectedFile)`